controllers/cleanup.go

```diff
@@ -0,0 +1,34 @@
+// Copyright Contributors to the Open Cluster Management project
+package controllers
+
+import (
+	"context"
+	"time"
+
+	searchv1alpha1 "github.com/stolostron/search-v2-operator/api/v1alpha1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	addonapiv1alpha1 "open-cluster-management.io/api/addon/v1alpha1"
+)
+
+func (r *SearchReconciler) deleteClusterManagementAddon(instance *searchv1alpha1.Search) error {
+	log.V(2).Info("Deleting ClusterManagementAddon search-collector")
+	cma := &addonapiv1alpha1.ClusterManagementAddOn{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ClusterManagementAddon",
+			APIVersion: "addon.open-cluster-management.io",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "search-collector",
+			Namespace: instance.GetNamespace(),
+		},
+	}
+	err := r.Delete(context.TODO(), cma)
+	if err != nil && !errors.IsNotFound(err) {
+		log.Error(err, "Failed to delete ClusterManagementAddon", "name", cma)
+		return err
+	}
+	time.Sleep(1 * time.Second)
```

Comment:
Why do we need to sleep?

Reply:
If it's for the retry logic, maybe only sleep if there's and error?

Reply:
Had that only as a safety measure - Did not want to start a reconcile on a instance which is SET to delete .I am removing it for now , if we see any issues , we can check again.